lib/travis/api/v3/services/build/restart.rb

```diff
@@ -6,9 +6,8 @@ def run
       access_control.permissions(build).restart!
 
       build.clear_debug_options!
-      if query.restart(access_control.user)
-        accepted(build: build, state_change: :restart)
-      end
+      return abuse_detected unless query.restart(access_control.user)
```

Comment:
It is not always because of abuse detection that a build may not restart. But thanks for the PR, I will fix the rest <3